### ISSUE TITLE
simplified SDC crashes with "too many subcycles" on GPU

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/Castro#752]
Build the `flame_wave` problem as:
```
make COMP=pgi USE_SIMPLIFIED_SDC=TRUE USE_CUDA=TRUE USE_MPI=FALSE -j 20
```
and run using the attached inputs / probin (this can run on a single GPU, e.g., groot).  You get almost immediately:
```
Advance was unsuccessful; proceeding to a retry.

  Subcycle completed


  Timestep 4.554566628e-10 rejected at level 1.
  Performing a retry, with subcycled timesteps of maximum length dt = 2.277283314e-10


  The subcycle mechanism requested 32 subcycled timesteps, which is larger than the maximum of 16.
  If you would like to override this, increase the parameter castro.max_subcycles.
amrex::Abort::0::Error: too many subcycles. !!!
SIGABRT
See Backtrace.0 file for details
```
[inputs.noboost.500Hz.txt](https://github.com/AMReX-Astro/Castro/files/4149101/inputs.noboost.500Hz.txt)
[probin.noboost.500Hz.txt](https://github.com/AMReX-Astro/Castro/files/4149102/probin.noboost.500Hz.txt)

**Comment by zingale** [comment]
Note: it runs fine on CPU.

**Comment by zingale** [comment]
fixed in #753
